core/frontend/src/Viewport.ts

```diff
@@ -816,6 +816,21 @@ export abstract class Viewport implements IDisposable {
   public changeBackgroundMapProps(props: BackgroundMapProps): void {
     this.displayStyle.changeBackgroundMapProps(props);
   }
+  /** Get a tile tree ref will load a tree that returns geometry rather than graphics.  Currently only the return of Polyface geometry from reality tile
+   * or map tile trees (with terrain enabled) are supported, calling [[getTileTreeRef]] on other tile trees will return undefined.
+   * Tile geometry can be  by used indirectly by methods such as [[RealityTileTree.drapeLineString]] or by calling [[RealityTileTree.collectTiles]] and
+   * iterating through[[RealityTileCollector.acceptedGeometry]]
+   */
+  public getGeometryTreeRef(modelId?: Id64String): TileTreeReference | undefined {

```

Comment:
Why is this on Viewport and why is it `@public`? Seems like the logic for finding the desired tile tree ref belongs in your tool.

Reply:
It should be @alpha (I'll fix) -- but it belongs here to hide the somewhat obtuse process of finding a treeRef for a modelId -- and making it so that we don't need to expose treeRef.creatGeometryTreeRef.   Why force every one to have this tricky code in their own application?

Reply:
Multiple tree references can exist with the same model Id.

Reply:
Multiple reality or map tree refs with same model id would not exist on the same viewport...   But point taken.   My implementation matches the civil groups requirements but there should be a way to create a geometry tree reference unassociated with any viewport so that you could query cesium terrain or reality tile set that is not displayed at all.   That is a somewhat bigger job though...and I'm not going to tackle it with this PR.